source/client.c

```diff
@@ -651,9 +816,10 @@ uint16_t aws_mqtt_client_connection_subscribe(
 handle_error:
 
     if (task_arg) {
-        if (task_arg->filter) {
-            aws_string_destroy((void *)task_arg->filter);
-        }
+
+        aws_string_destroy(task_topic->filter);
+        aws_mem_release(connection->allocator, task_topic);
+
         aws_mem_release(connection->allocator, task_arg);
```

Comment:
all the things you're freeing here may have failed to allocate in the first place, and our deleters don't expect`null`